Source/FieldSolver/FiniteDifferenceSolver/HybridPICModel/ExternalVectorPotential.cpp

```diff
@@ -210,6 +211,14 @@ ExternalVectorPotential::InitData ()
 
         amrex::Gpu::streamSynchronize();
 
+        if (m_do_clean_divA && warpx.grid_type == GridType::Collocated) {
```

Comment:
This is only called during the initialization, right? If so, why not also do the divergence cleaning when the grid type is staggered?
I guess in principle, this is only needed when A has time dependence. Could the check be for that instead?

Reply:
I only am doing it now when non-staggered because the staggered version is set up to clean on face-centered staggering and the A field is an edge-centered field. The solver for a staggered B field needs a cell-centered operator to work. In the collocated version I use a nodal solver, which I would need for the face-centered case.

I think this would need a bit more testing for the cleaning of A when staggered. I will modify it and give it a shot.